sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala

```diff
@@ -1312,6 +1328,19 @@ class RapidsShuffleInternalManagerBase(conf: SparkConf, val isDriver: Boolean)
 
   override def unregisterShuffle(shuffleId: Int): Boolean = {
     unregisterGpuShuffle(shuffleId)
+    shuffleBlockResolver match {
+      case isbr: IndexShuffleBlockResolver =>
+        Option(taskIdMapsForShuffle.remove(shuffleId)).foreach { mapTaskIds =>
+          mapTaskIds.iterator.foreach { mapTaskId =>
+            isbr.removeDataByMap(shuffleId, mapTaskId)
+          }
+        }
+      case _ if taskIdMapsForShuffle.size() > 0 =>
+        logError("unregisterShuffle called with " +
+          s"unexpected resolver ${shuffleBlockResolver} and ${taskIdMapsForShuffle.size()} " +
+          s"blocks data to clean!")
+        taskIdMapsForShuffle.clear()
```

Comment:
I am confused by this. Why clear the tracking info when we see something we don't expect? And why do we leak the data in that case?

Reply:
You are right, throwing instead: https://github.com/NVIDIA/spark-rapids/pull/7125/commits/419b13b04eb7eeacff2fc4485241a53f890059bc. Spark only has the `SortShuffleManager` which hard-codes the `IndexShuffleBlockResolver` so this is very unlikely to happen unless we are dealing with a different distro, and we'd like this to fail loudly then.